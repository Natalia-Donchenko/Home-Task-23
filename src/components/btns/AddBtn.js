import { useDispatch } from "react-redux";
import { addTodo } from "../../store/todoSlice";

const AddBtn = ( { setState, state, text }) => {

  const dispatch = useDispatch();

  const add = () => {
    if(text === '') {
      setState({...state});
    };
    
    dispatch(addTodo({newText: text}));
    setState({...state, text: ''});
  }
  return (
    <button 
      type='button' 
      className="addTask-btn"
      onClick={add}
    >
      add
    </button>
  );
};

export default AddBtn;