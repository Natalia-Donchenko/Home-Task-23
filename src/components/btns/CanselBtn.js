const CanselBtn = ( { setEditing, isEditing }) => {

  const canselEdit = () => {
    setEditing(!isEditing);
  };

  return (
    <button
      type="button"
      className="editing-btn addTask-btn"
      onClick={canselEdit}
      id="cbtn"
    >
      cansel
    </button>
  );
};

export default CanselBtn;