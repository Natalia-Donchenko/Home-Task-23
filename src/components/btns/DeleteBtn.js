import { useDispatch } from "react-redux";
import { deleteTodo } from "../../store/todoSlice";

const DeleteBtn = ({ id }) => {

  const dispatch = useDispatch()  

  return (
    <button
      type="button"
      className="optional-btn"
      onClick={() => dispatch(deleteTodo({id}))}
    >
      delete
    </button>
  )
};

export default DeleteBtn;