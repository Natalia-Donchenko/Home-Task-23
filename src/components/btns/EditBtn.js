const EditBtn = ({ done, id, text, setEditing, setState, state }) => {

  const onEditToggle = ( id, text) => {
    setEditing(true);
    setState({
      ...state, id, text
    });
  };

  return (
    <button
      type="button"
      className={done ? "btn-done" : "optional-btn"}
      onClick={() =>onEditToggle(id, text)}
    >
      edit
    </button>
  )
};

export default EditBtn;