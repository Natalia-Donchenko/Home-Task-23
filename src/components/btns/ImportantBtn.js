import { useDispatch } from "react-redux";
import { isImportant } from "../../store/todoSlice";

const ImportantBtn = ({ id, done, important}) => {

  const dispatch = useDispatch();

  return (
    <button
      type="button"
      onClick={() => dispatch(isImportant({id}))}
      className={ done ? "btn-done": "" || important ? "imp-btn" : "optional-btn"}
    >
      important
    </button>
  );
};

export default ImportantBtn;