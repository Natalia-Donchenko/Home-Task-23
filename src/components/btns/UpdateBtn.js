import { useDispatch } from "react-redux";
import { editTodo } from "../../store/todoSlice";

const UpdateBtn = ({ text, setState, state, id, setEditing }) => {
  
  const dispatch = useDispatch();

  const edit = () =>{
    if (text === '') {
      setState({...state});
      return;
    }
    
    dispatch((editTodo({text, id})));
    setEditing(false);
 };

  return (
    <button 
      type='button' 
      className="editing-btn addTask-btn" 
      onClick={edit}
    >
      update
    </button>
  );
};

export default UpdateBtn;