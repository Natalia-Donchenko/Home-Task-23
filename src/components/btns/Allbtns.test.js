import { Provider} from "react-redux"
import AddBtn from "./AddBtn"
import CanselBtn from "./CanselBtn"
import DeleteBtn from "./DeleteBtn"
import EditBtn from "./EditBtn"
import UpdateBtn from "./UpdateBtn"
import ImportantBtn from "./ImportantBtn"
import { render, cleanup } from "@testing-library/react"
import store from "../../store"

afterEach(cleanup);

describe("Button Add", () => {

  const { container } = render(
    <Provider store={store}>
      <AddBtn title="AddBtn" />
    </Provider>
  );

    test("should render Add btn", () => {
      expect(container).toMatchSnapshot();
    });
});


describe("Button Cansel", () => {

  const { container } = render(
    <Provider store={store}>
      <CanselBtn  title="CanselBtn" />
    </Provider>
  )

  it("should render Cansel btn", () => {
    expect(container).toMatchSnapshot();
  });
});


describe("Button Delete", () => {

  const { container } = render(
    <Provider store={store}>
      <DeleteBtn  title="DeleteBtn" />
    </Provider>
  );

  it("should render Delete btn", () => {
    expect(container).toMatchSnapshot();
  });
});


describe("Button Edit", () => {
  const { container } = render(
    <Provider store={store}>
      <EditBtn  title="EditBtn" />
    </Provider>
  );

  it("should render Edit btn", () => {
    expect(container).toMatchSnapshot();
  });
});


describe("Button Important", () => {
  const { container } = render(
    <Provider store={store}>
      <ImportantBtn  title="ImportantBtn" />
    </Provider>
  );

  it("should render Important btn", () => {
    expect(container).toMatchSnapshot();
  });
});


describe("Button Update", () => {

  const { container } = render(
    <Provider store={store}>
      <UpdateBtn  title="UpdateBtn" />
    </Provider>
  );

  it("should render Update btn", () => {
    expect(container).toMatchSnapshot();
  });
});