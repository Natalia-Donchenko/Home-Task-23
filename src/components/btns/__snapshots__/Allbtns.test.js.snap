// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 'Add' should render Add btn 1`] = `
<div>
  <button
    class="addTask-btn"
    type="button"
  >
    add
  </button>
</div>
`;

exports[`Button 'Cansel' should render Cansel btn 1`] = `<div />`;

exports[`Button 'Delete' should render Delete btn 1`] = `<div />`;

exports[`Button 'Edit' should render Edit btn 1`] = `<div />`;

exports[`Button 'Important' should render Important btn 1`] = `<div />`;

exports[`Button 'Update' should render Update btn 1`] = `<div />`;

exports[`Button Add should render Add btn 1`] = `
<div>
  <button
    class="addTask-btn"
    type="button"
  >
    add
  </button>
</div>
`;

exports[`Button Add' should render Add btn 1`] = `
<div>
  <button
    class="addTask-btn"
    type="button"
  >
    add
  </button>
</div>
`;

exports[`Button Cansel should render Cansel btn 1`] = `<div />`;

exports[`Button Delete should render Delete btn 1`] = `<div />`;

exports[`Button Edit should render Edit btn 1`] = `<div />`;

exports[`Button Important should render Important btn 1`] = `<div />`;

exports[`Button Update should render Update btn 1`] = `<div />`;
